### DISCUSSION TITLE
.NET Framework 4.8 Not Supported for Some APIs

### DISCUSSION THREAD

**Discussion by khdc** [microsoft/CsWin32#868] [Q&A] [answered]
I have a WPF library project that multi-targets: net48, net6-windows, and net7-windows.

I noticed that APIs/types like HWND, WINDOW_STYLE, GetDeviceCaps, GetSystemMetrics, MonitorFromWindow, etc., show as not supported on net48.

Is there a reason why?

Should I switch back to the olden way of doing PInvoke?

**Comment by khdc** [comment]
The error occured when I removed this line from the csproj file as per the readme.md:

`<IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>`

When I added it back in the error went away. I'm not sure why.

**Comment by AArnott** [comment, marked as answer]
These APIs _are_ supported on net48. There must be something else going on in your project that is hiding HWND and the like.
Check out this trivial project:
[CsWin32Sandbox.zip](https://github.com/microsoft/CsWin32/files/10843856/CsWin32Sandbox.zip)

It compiles its code that includes and uses HWND for both TFMs just fine:

```
Build started...
1>------ Build started: Project: CsWin32Sandbox, Configuration: Debug Any CPU ------
1>C:\Users\andarno\source\repos\CsWin32Sandbox\Program.cs(5,6,5,10): warning CS0219: The variable 'hWND' is assigned but its value is never used
1>CsWin32Sandbox -> C:\Users\andarno\source\repos\CsWin32Sandbox\bin\Debug\net472\CsWin32Sandbox.exe
1>Done building project "CsWin32Sandbox.csproj".
1>C:\Users\andarno\source\repos\CsWin32Sandbox\Program.cs(5,6,5,10): warning CS0219: The variable 'hWND' is assigned but its value is never used
1>CsWin32Sandbox -> C:\Users\andarno\source\repos\CsWin32Sandbox\bin\Debug\net7.0\CsWin32Sandbox.dll
1>Done building project "CsWin32Sandbox.csproj".
========== Build: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
========== Build started at 1:09 PM and took 00.362 seconds ==========
```

**Reply by khdc** [reply]
It's weird the problem went away when I added back the following to the csproj file. Maybe it was just a coincidence, or Visual Studio was having issues. 

`<IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>`
